packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1680,6 +1856,129 @@ export class MergeTree {
 				saveIfLocal(newSegment);
 
 				insertPos += newSegment.cachedLength;
+
+				if (!this.options?.mergeTreeEnableObliterate) {
```

Comment:
filed https://dev.azure.com/fluidframework/internal/_workitems/edit/4010 to improve this